### PR TITLE
sorts map and vector keyword entries in router-state-maintainer

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- sorts map and vector keyword entries in router-state-maintainer
- uses threading macro in do-health-checks

## Why are we making these changes?

"We" like things ordered lexicographically :)

**Note**: [Hiding whitespaces in the diff](https://github.com/twosigma/waiter/pull/419/files?w=1) might make it easier to review.

### CODE REVIEW

File: waiter/src/waiter/scheduler.clj [hunk 1]
```diff
@@ -447,14 +447,15 @@
...
       (if-not service
         service->service-instances'
-        (let [active-instances (doall (map async/<!! active-instances))]
-          (recur rest (assoc service->service-instances'
-                        service
-                        (assoc instances :active-instances active-instances))))))))
+        (recur rest
+               (->> active-instances
+                    (map async/<!!)
+                    doall
```
Comment (DaoWen) [inline waiter/src/waiter/scheduler.clj]: How about `mapv` in place of `map` + `doall`?

Comment (shamsimam) [reply waiter/src/waiter/scheduler.clj]: Done.

Comment (shamsimam) [reply waiter/src/waiter/scheduler.clj]: Removing this change as it breaks the code.

**Comment by shamsimam** [comment]
@DaoWen  Build is green

**Pull request merged**